### DISCUSSION TITLE
TSL / SSL error

### DISCUSSION THREAD

**Discussion by KojieKabuto** [microsoft/aisdkforsapabap#23] [Q&A]
Hello,

I'm running ZP_MSAISDKDEMO_MODELS_OAI (got an openAI key already)  but keeps dumping. (client communication failure). I checked SMICM and it seems to be encountering TSL/SSL error. Are there any documentation on how to solve this issue? Our new basis guy does not seem to know how to solve this. We're running SAP 7.5, on a VPN.

Thank you.

**Comment by gopalnair** [comment]
Hi @KojieKabuto ,
   It seems like the certificate is not installed properly in transaction: **STRUST**
This problem is un-usual, since most SAP systems will have the common root certificate installed. 

### To solve this, please do the following: ###

1. Open the URL : [https://api.openai.com/](https://api.openai.com/)  *(NOTE : You will see a JSON error... its OK. We are interested in downloading the certificate)*
1. Download the certificates from this site, as [shown in this video](https://youtu.be/BnFtloLM52Y?t=111) *(NOTE : In the video, we are seeing how to download the certificate from github - but its the same procedure to download the certificate from Open AI site as well)*. 
1. Open transaction STRUST, and install this in "Standard" area *(NOTE : The video shows the github certificate being installed in Anonymous section. You need to install the Open AI certificate in the "standard" node though. )*


Once the above steps are done, you should be able to run the demo program, and communication should be successful - assuming, there is no other organization policy preventing external communications.

Please let me know how it went, and if the issue is resolved.

Best Regards,
Gopal Nair.
